library/src/main/kotlin/com/connectrpc/protocols/GRPCInterceptor.kt

```diff
@@ -161,12 +161,20 @@ internal class GRPCInterceptor(
                     onCompletion = { result ->
                         val trailers = result.trailers
                         val completion = completionParser.parse(emptyMap(), trailers)
-                        val connectError = grpcCompletionToConnectError(completion, serializationStrategy, result.error)
-                        StreamResult.Complete(
-                            code = connectError?.code ?: Code.OK,
-                            error = connectError,
-                            trailers = trailers,
-                        )
+                        if (completion != null) {
+                            val exception = completion.toConnectExceptionOrNull(
+                                serializationStrategy,
+                                result.cause,
+                            )
+                            StreamResult.Complete(
+                                code = exception?.code ?: Code.OK,
+                                cause = exception,
+                                trailers = trailers,
+                            )
+                        } else {
+                            val exception = ConnectException(Code.UNKNOWN)
```

Comment:
To resolve the test failure, this else condition should change to:

```kotlin
return@fold result
```

This way we'll preserve errors like `DEADLINE_EXCEEDED`.

Reply:
So result is already instantiated in some cases (with a code like DEADLINE_EXCEEDED and a cause). Some of the tests rely on this. Alternatively to doing `return@fold`. you could just have `result` here as the return value.

Reply:
I'm tracking now. Thanks for the explaination.